src/Features/Core/Portable/ChangeSignature/Parameter.cs

```diff
@@ -21,28 +25,63 @@ public ExistingParameter(IParameterSymbol param)
             Symbol = param;
         }
 
-        public override bool HasExplicitDefaultValue => Symbol.HasExplicitDefaultValue;
+        public override bool HasDefaultValue => Symbol.HasExplicitDefaultValue;
         public override string Name => Symbol.Name;
     }
 
     internal sealed class AddedParameter : Parameter
     {
-        public AddedParameter(ITypeSymbol type, string typeNameDisplayWithErrorIndicator, string parameter, string callSiteValue)
+        public AddedParameter(
+            ITypeSymbol type,
+            string typeName,
+            string name,
+            string callSiteValue,
+            bool isRequired = true,
+            string defaultValue = "",
+            bool useNamedArguments = false,
+            bool isCallsiteOmitted = false,
+            bool isCallsiteError = false,
+            bool typeBinds = true)
         {
             Type = type;
-            TypeNameDisplayWithErrorIndicator = typeNameDisplayWithErrorIndicator;
-            ParameterName = parameter;
+            TypeBinds = typeBinds;
+            TypeName = typeName;
+            Name = name;
             CallSiteValue = callSiteValue;
+
+            IsRequired = isRequired;
+            DefaultValue = defaultValue;
+            IsCallsiteError = isCallsiteError;
+            IsCallsiteOmitted = isCallsiteOmitted;
+            UseNamedArguments = useNamedArguments;
+
+            if (IsCallsiteError)
+            {
+                CallSiteValue = FeaturesResources.ChangeSignature_NewParameterIntroduceTODOVariable;
+            }
+            else if (isCallsiteOmitted)
+            {
+                CallSiteValue = FeaturesResources.ChangeSignature_NewParameterOmitValue;
+            }
+            else
+            {
+                CallSiteValue = callSiteValue;
+            }
         }
 
-        public ITypeSymbol Type { get; set; }
-        public string ParameterName { get; set; }
-        public string CallSiteValue { get; set; }
+        public override string Name { get; }
+        public override bool HasDefaultValue => !string.IsNullOrWhiteSpace(DefaultValue);
 
-        public override bool HasExplicitDefaultValue => false;
-        public override string Name => ParameterName;
+        public ITypeSymbol Type { get; }
+        public string TypeName { get; }
+        public bool TypeBinds { get; }
+        public string CallSiteValue { get; }
 
-        public string TypeNameDisplayWithErrorIndicator { get; set; }
+        public bool IsRequired { get; }
+        public string DefaultValue { get; }
+        public bool UseNamedArguments { get; }
+        public bool IsCallsiteOmitted { get; }
+        public bool IsCallsiteError { get; }
```

Comment:
doc these.

Reply:
Renamed to IsCallsiteTodo and also doc'd.